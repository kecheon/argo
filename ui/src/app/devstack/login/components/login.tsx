import * as React from 'react';
import { useState } from 'react';
import {Form} from 'react-bootstrap';
import {uiUrl} from '../../../shared/base';
import { UserState } from '../../classes/current-user';
import {UserService} from '../../services/user-service';
import axios from 'axios';

require('./login.scss');

export interface LoginForm {
    username: string;
    password: string;
    domainId: string;
}

export default () => {
    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');
    const [domainId, setDomainId] = useState('default');
    const [user, setUser] = useState(null);
    const service = new UserService();
    const handleSubmit = async (e: any) => {
        e.preventDefault();
        const endpoint = 'http://localhost:3000';
        const credential = {
            username,
            domainId,
            password
        }
        const res = await axios.post(`${endpoint}/account/login`, credential);
        if (res.status === 200) {
            console.log(res.data);
            const { name, jwtToken, roles }= res.data;

            const currentUser: UserState = {
                isLoggedIn: true,
                username: name,
                role: {name: roles[0], level: 0},
                accessToken: jwtToken
            }
            localStorage.setItem('user', JSON.stringify(currentUser));
            localStorage.setItem('accessToken', currentUser.accessToken);
            setUser(currentUser);
            if (currentUser.role.level <= 0) {
                document.location.href = uiUrl('overview');
            } else {
                document.location.href = uiUrl('workflows');
            }

        } else {
            alert(`login Error ${username} ${password}`);
        }
        // if (res.data.status === 'success') {
        //     // TODO
        //     // set role of this user
        //     const { roles }= res.data.user;

        //     const currentUser: UserState = {
        //         isLoggedIn: true,
        //         username: 'admin',
        //         role: {name: roles[0], level: 0},
        //         accessToken: 'jwtToken_if_exists'
        //     }
        //     localStorage.setItem('user', JSON.stringify(currentUser));
        //     localStorage.setItem('accessToken', currentUser.accessToken);
        //     setUser(currentUser);
        //     if (currentUser.role.level <= 0) {
        //         document.location.href = uiUrl('overview');
        //     } else {
        //         document.location.href = uiUrl('workflows');
        //     }

        // } else {
        //     alert(`login Error ${username} ${password}`);
        // }
    }
   
    return (
        <div className='login'>
            <div className='login__box'>
                <form onSubmit={ handleSubmit }>
                    <div className='login__logo width-control'>
                        <img className='logo-image' src='assets/images/devstack/logo.png' alt='devStack' />
                    </div>
                    <div className='argo-form-row'>
                        <Form.Group controlId='formBasicUsername'>
                            <Form.Label>User Name*</Form.Label>
                            <Form.Control type='text' placeholder='Enter username'
                                value={ username }
                                name='username'
                                onChange={(e: { target: { value: React.SetStateAction<string>; }; }) => setUsername(e.target.value)}/>
                        </Form.Group>
                    </div>
                    <div className='argo-form-row'>
                        <Form.Group controlId='formBasicDomainId'>
                            <Form.Label>Domain ID*</Form.Label>
                            <Form.Control type='text' placeholder='Enter domain id'
                                value={ domainId }
                                name='domainId'
                                onChange={(e: { target: { value: React.SetStateAction<string>; }; }) => setDomainId(e.target.value)}/>
                        </Form.Group>
                    </div>
                    <div className='argo-form-row'>
                        <Form.Group controlId='formBasicPassword1'>
                            <Form.Label>Password</Form.Label>
                            <Form.Control type='password' placeholder='Password'
                                value={ password }
                                name='password'
                                onChange={(e: { target: { value: React.SetStateAction<string>; }; }) => setPassword(e.target.value)}/>
                        </Form.Group>
                    </div>
                    <div className='login__form-row'>
                        <button className='argo-button argo-button--base argo-button--full-width argo-button--xlg' type='submit'>
                            Sign In
                        </button>
                    </div>
                    {/* <div className='login__form-row'>
                        <button className='argo-button argo-button--base argo-button--full-width argo-button--xlg' 
                            type='button'
                            onClick={() => document.location.href=uiUrl('register')}
                        >
                            Sign Up
                        </button>
                    </div> */}
                    <div className='login__footer'>
                        <a href='https://argoproj.io' target='_blank'>
                            <img className='logo-image' src='assets/images/argologo.svg' alt='argo' />
                        </a>
                    </div>
                </form>
            </div>
        </div>
    );
};