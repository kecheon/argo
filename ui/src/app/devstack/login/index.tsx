import Login from './components/loginSession';

export default {
    component: Login
};
