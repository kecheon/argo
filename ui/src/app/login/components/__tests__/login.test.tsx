import '@testing-library/jest-dom/extend-expect';
import {render} from '@testing-library/react';
import * as React from 'react';
import {Login} from '../login';

describe('<Login /> tests', () => {
    it('should render without crash', () => {
        const {container} = render(<Login />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
