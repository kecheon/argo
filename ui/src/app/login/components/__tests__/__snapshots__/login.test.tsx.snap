// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> tests should render without crash 1`] = `
<div
  class="page page--has-toolbar"
>
  <div
    class="page__top-bar"
  >
    <div>
      <div
        class="top-bar"
      >
        <div
          class="row"
        >
          <div
            class="columns top-bar__left-side"
          >
            <div
              class="top-bar__breadcrumbs"
            >
              <span
                class="top-bar__breadcrumbs-last-item"
              >
                Login
              </span>
            </div>
          </div>
          <div
            class="top-bar__title text-truncate top-bar__right-side"
          >
            Login
          </div>
        </div>
      </div>
      <div
        class="top-bar row"
      >
        <div
          class="columns small-9 top-bar__left-side"
        />
        <div
          class="columns small-3 top-bar__right-side"
        />
      </div>
    </div>
  </div>
  <div
    class="page__content-wrapper"
  >
    <div
      class="argo-container"
    >
      <div
        class="white-box"
      >
        <h3>
          <i
            class="fa fa-shield-alt"
          />
           Login
        </h3>
        <p>
          It may not be necessary to be logged in to use Argo Workflows, it depends on how it is configured.
        </p>
        <p>
          <a
            href="https://argoproj.github.io/argo/argo-server-auth-mode/"
          >
            Learn more
          </a>
          .
        </p>
      </div>
      <div
        class="row"
      >
        <div
          class="columns small-4"
        >
          <p>
            If your organisation has configured 
            <b>
              single sign-on
            </b>
            :
          </p>
          <div>
            <button
              class="argo-button argo-button--base-o"
            >
              <i
                class="fa fa-sign-in-alt"
              />
               Login
            </button>
          </div>
        </div>
        <div
          class="columns small-4"
        >
          <p>
            If your organisation has configured 
            <b>
              client authentication
            </b>
            , get your token using 
            <code>
              argo auth token
            </code>
             and paste in this box:
          </p>
          <div>
            <textarea
              cols="32"
              id="token"
              rows="8"
            />
          </div>
          <div>
            <button
              class="argo-button argo-button--base-o"
            >
              <i
                class="fa fa-sign-in-alt"
              />
               Login
            </button>
          </div>
        </div>
        <div
          class="columns small-4"
        >
          <div>
            <p>
              Something wrong? Try logging out and logging back in:
            </p>
            <button
              class="argo-button argo-button--base-o"
            >
              <i
                class="fa fa-sign-out-alt"
              />
               Logout
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
